src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -151,17 +154,63 @@ void ClusterTaskManager::TasksUnblocked(const std::vector<TaskID> ready_ids) {
   }
 }
 
+void ClusterTaskManager::HandleTaskFinished(std::shared_ptr<WorkerInterface> worker) {
+  cluster_resource_scheduler_->SubtractCPUResourceInstances(
+                                                        worker->GetBorrowedCPUInstances());
+  cluster_resource_scheduler_->FreeLocalTaskResources(worker->GetAllocatedInstances());
+  worker->ClearAllocatedInstances();
+}
+
+bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
```

Comment:
Should we also have a unit test task for task cancellation? Just to check that the maps are cleaned up, etc.